s3fs/core.py

```diff
@@ -788,6 +856,35 @@ def info(self):
         """ File information about this path """
         return self.s3.info(self.path)
 
+    def metadata(self, refresh=False):
+        """ Return metadata of file.
+        See :func:`~s3fs.S3Filesystem.metadata`.
+
+        Metadata is cached unless `refresh=True`.
+        """
+        return self.s3.metadata(self.path, refresh)
+
+    def getxattr(self, xattr_name):
+        """ Get an attribute from the metadata.
+        See :func:`~s3fs.S3Filesystem.getxattr`.
+
+        Examples
+        --------
+        >>> mys3file.getxattr('attribute_1')  # doctest: +SKIP
+        'value_1'
+        """
+        return self.s3.getxattr(self.path, xattr_name)
+
+    def setxattr(self, **kwargs):
+        """ Set metadata.
+        See :func:`~s3fs.S3Filesystem.setxattr`.
+
+        Examples
+        --------
+        >>> mys3file.setxattr(attribute_1='value1', attribute_2='value2')  # doctest: +SKIP
+        """
+        return self.s3.setxattr(self.path, **kwargs)
```

Comment:
I don't suppose this can work for a file open for writing? Perhaps in that case, we need to store the metadata and apply upon close. For a file in append mode, we need to store the original metadata and apply to the final file.